### PR TITLE
helm: move migrations to initContainer

### PR DESCRIPTION
**Pull request by AurelienGasser**

### CODE REVIEW

File: charts/substra-backend/templates/deployment-backend.yaml [hunk 1]
```diff
@@ -37,9 +117,9 @@ spec:
           {{- end }}
           command: ["/bin/bash"]
           {{- if eq .Values.backend.settings "prod" }}
-          args: ["-c", "python manage.py migrate && python3 manage.py collectstatic --noinput && uwsgi --http :8000 --module backend.wsgi --static-map /static=/usr/src/app/backend/statics --master --processes 4 --threads 2 --need-app --env DJANGO_SETTINGS_MODULE=backend.settings.server.{{ .Values.backend.settings }} "]
```
Comment (ClementGautier) [inline charts/substra-backend/templates/deployment-backend.yaml]: Can you also move the collectstatic command in the init-container? I could help restart time of the pod (you might need to share a volume for the static files, you can use an emptyDir)

File: charts/substra-backend/templates/deployment-backend.yaml [hunk 2]
```diff
@@ -29,6 +29,86 @@ spec:
         - name: {{ . }}
       {{- end }}
       {{- end }}
+      initContainers:
+        - name: {{ template "substra.name" . }}-backend-init-postgres
+          image: "{{ .Values.backend.image.repository }}:{{ .Values.backend.image.tag }}"
```
Comment (ClementGautier) [inline charts/substra-backend/templates/deployment-backend.yaml]: Use a smaller image for that, you can try `dockerize` for example:
```
dockerize -wait tcp://$DATABASE_HOST:$DATABASE_PORT
```

**Comment by GuillaumeCisco** [comment]
I used your work for creating this PR: https://github.com/GuillaumeCisco/substra-backend/pull/1/files

**Comment by AurelienGasser** [comment]
Closed in favor of https://github.com/GuillaumeCisco/substra-backend/pull/1/files

**Pull request closed without merging**